### PR TITLE
refactor(cypress): Migrate to new logo & update extensions

### LINKED ISSUE #2910: [Icon Request]: Cypress Logo Rebrand
👋🏻 Hello! I'm excited to add the new Cypress logo!
We've rebranded, added a new config file type, and added a new extension to avoid conflicts with Jest/Mocha specs.
https://twitter.com/Cypress_io

- [x] I'm sure this issue is not a _duplicate_.
- [x] I'm submitting an icon request:
  - Type: `extension`, `folder`
  - Icon Name: `cypress`
  - Sample original Icon:  cypress-icon
  - Extensions: `.cy.*` (Just like how `.spec.*` works)
  - Filenames:
    - `cypress.config.{js,ts}`
    - `cypress.json`

### PR DESCRIPTION
**Pull request by sasial-dev**
_**Fixes #2910**_

**Changes proposed:**

Update to new cypress logo & add new file extensions

### CODE REVIEW

**Comment by sasial-dev** [comment]
Cross post from the inital issue:
> I've created a PR with the new logo, although some help would be appriciated with the folder icons (I'm not the greatest designer!)

File: src/iconsManifest/supportedExtensions.ts [hunk 1]
```diff
@@ -1182,7 +1182,9 @@ export const extensions: IFileCollection = {
     },
     {
       icon: 'cypress',
-      extensions: [`cypress.json`, `cypress.env.json`],
+      extensions: ['cypress.json', 'cypress.env.json'],
+      filenamesGlob: ['cypress.config'],
+      extensionsGlob: ['js', 'ts'],
```
Comment (JessicaSachs) [inline src/iconsManifest/supportedExtensions.ts:1187]:
```suggestion
      extensionsGlob: ['js', 'ts', 'mjs', 'cjs'],
```

**Review by JessicaSachs — COMMENTED** [review]
Do we need these lil extensions, too? `cypress.config.mjs` is something we handle.

File: src/iconsManifest/supportedExtensions.ts (hunk 1 above)
Comment (sasial-dev) [reply src/iconsManifest/supportedExtensions.ts:1187]: Yep, we do.
Good catch!

**Comment by sasial-dev** [comment]
Superseaded by #2982

**Pull request closed without merging**